lib/Conversion/SimToSV/SimToSV.cpp

```diff
@@ -144,20 +163,34 @@ struct SimToSVPass : public LowerSimToSVBase<SimToSVPass> {
     auto circuit = getOperation();
     MLIRContext *context = &getContext();
 
-    ConversionTarget target(*context);
-    target.addIllegalDialect<SimDialect>();
-    target.addLegalDialect<sv::SVDialect>();
-    target.addLegalDialect<hw::HWDialect>();
-    target.addLegalDialect<seq::SeqDialect>();
-    target.addLegalDialect<comb::CombDialect>();
-
-    RewritePatternSet patterns(context);
-    patterns.add<PlusArgsTestLowering>(context);
-    patterns.add<PlusArgsValueLowering>(context);
-    patterns.add<SimulatorStopLowering<sim::FinishOp, sv::FinishOp>>(context);
-    patterns.add<SimulatorStopLowering<sim::FatalOp, sv::FatalOp>>(context);
-    if (failed(applyPartialConversion(circuit, target, std::move(patterns))))
-      signalPassFailure();
+    SimConversionState state;
+    auto lowerModule = [&](hw::HWModuleOp module) {
+      ConversionTarget target(*context);
+      target.addIllegalDialect<SimDialect>();
+      target.addLegalDialect<sv::SVDialect>();
+      target.addLegalDialect<hw::HWDialect>();
+      target.addLegalDialect<seq::SeqDialect>();
+      target.addLegalDialect<comb::CombDialect>();
+
+      RewritePatternSet patterns(context);
+      patterns.add<PlusArgsTestLowering>(context, state);
+      patterns.add<PlusArgsValueLowering>(context, state);
+      patterns.add<SimulatorStopLowering<sim::FinishOp, sv::FinishOp>>(context,
+                                                                       state);
+      patterns.add<SimulatorStopLowering<sim::FatalOp, sv::FatalOp>>(context,
+                                                                     state);
+      return applyPartialConversion(module, target, std::move(patterns));
+    };
+
+    if (failed(mlir::failableParallelForEach(
+            context, circuit.getOps<hw::HWModuleOp>(), lowerModule)))
+      return signalPassFailure();
+
+    if (state.used_SYNTHESIS && !circuit.lookupSymbol("SYNTHESIS")) {
```

Comment:
Please use symbol table instead of Operation::lookupSymbol to avoid O(N^2) behaivor.

Reply:
Nvm, it's ran only once so don't need to use symbol table.